sdk/identity/identity/migration-v1-v2.md

```diff
@@ -0,0 +1,309 @@
+  - [Changes to all credentials in general](#changes-to-all-credentials-in-general)
+  - [Changes to the interactive credentials](#changes-to-the-interactive-credentials)
+  - [Changes to ClientSecretCredential and ClientCertificateCredential](#changes-to-clientsecretcredential-and-clientcertificatecredential)
+  - [Changes to the ManagedIdentityCredential](#changes-to-the-managedidentitycredential)
+  - [Changes to the InteractiveBrowserCredential](#changes-to-the-interactivebrowsercredential)
+  - [Changes to the AzureCliCredential](#changes-to-the-azureclicredential)
+  - [Changes to the DeviceCodeCredential](#changes-to-the-devicecodecredential)
+  - [Changes to the VisualStudioCodeCredential](#changes-to-the-visualstudiocodecredential)
+- [Plugins](#plugins)
+  - [Persistent token cache plugin](#persistent-token-cache-plugin)
+  - [VisualStudioCodeCredential plugin](#visualstudiocodecredential-plugin)
+- [Troubleshooting](#troubleshooting)
+- [Provide feedback](#provide-feedback)
+- [Contributing](#contributing)
+
+## Node.js version support 
+
+While the last `@azure/identity` version to support Node.js 8 is 1.3.0, `@types/node` has been updated to version 12. For more details, read the [support policy](https://github.com/Azure/azure-sdk-for-js/blob/main/SUPPORT.md).
+
+## Dependency changes
+
+Version 2 of Azure Identity for JavaScript no longer includes native dependencies (neither ordinary, peer, nor optional dependencies). Previous distributions of `@azure/identity` included an optional dependency on `keytar`, which caused issues for some users in restrictive environments.
+
+Version 2 of Azure Identity for JavaScript now also depends on the latest available versions of `@azure/msal-common`, `@azure/msal-node`, and `@azure/msal-browser`. Our goal is to always be up-to-date with the MSAL versions.
+
+All credentials provided in version 2 of the Identity package for Node.js now make use of the latest `@azure/msal-node`. `InteractiveBrowserCredential` now uses the latest `@azure/msal-node` on Node.js and the latest `@azure/msal-browser` on the browser.
+
+Relying on the latest versions of MSAL comes with many benefits, but it has some considerations:
+
+- For every new instance of all of the updated credentials, now they will do a first request to discover relevant endpoint metadata information from Azure.
+- On the browser, the `InteractiveBrowserCredential` will now use the [Auth Code Flow](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-auth-code-flow) with [PKCE](https://tools.ietf.org/html/rfc7636) rather than [Implicit Grant Flow](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-implicit-grant-flow) to better support browsers with enhanced security restrictions. Read more on this in our [docs on Interactive Browser Credential](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/interactive-browser-credential.md).
```

Comment:
This reads like an implementation detail and does not tell the customer if they have to do something different when moving to v2 if they have been on the browser.

Reply:
I will update this with instructions on how to create a new app registration or update the app registration in use. That should be the only change required by users.